tests/rptest/clients/rpk.py

```diff
@@ -36,6 +36,14 @@ def __str__(self):
             self.id, self.leader, self.replicas, self.high_watermark,
             self.start_offset)
 
+    def __eq__(self, other):
+        if other is None:
+            return False
```

Comment:
where is the comparison to `None` made? @graphcareful

Reply:
https://github.com/redpanda-data/redpanda/blob/dev/tests/rptest/tests/wasm_topics_test.py#L49

Reply:
it seems like both output and test_cfg will be non-None in that condition

Reply:
```python
        if test_cfg == None:
            raise Exception("Input topic missing {}" % self.topic)
        test_cfg = sorted(test_cfg, key=rpk_partition_sort)

        # Ensure all materialized topics created and configs match inputs
        for topic in self.wasm_test_output():
            output = self._rpk_tool.describe_topic(topic)
            if output == None:
                raise Exception("Materialized topic missing {}" % topic)
            output = sorted(output, key=rpk_partition_sort)
            if output != test_cfg:
```

for both sides of != we raise if they are equal to None.

Reply:
I know it seems odd. The test was failing so i printed the contents of both lists and they were identical. It then passed when i added the conditional.

I chalked it up to some python oddities that i was unfamiliar with as I'm not a python ace.

I can debug further in following PRs to fully reintroduce WASM tests into CI.

Reply:
Also i wasn't saying the case could be one of these list would be None, it's if an element within the list i.e. an RpkPartition is None.

Reply:
you change `sorted(output, key=rpk_partition_sort)` to `filter(None, sorted(output, key=rpk_partition_sort))` to remove None. I think it will be error prone having the check for `None` in `def __eq__`.